iree/compiler/Dialect/HAL/Conversion/HALToVM/test/control_flow_ops.mlir

```diff
@@ -0,0 +1,21 @@
+// RUN: iree-opt -split-input-file -iree-convert-hal-to-vm %s | IreeFileCheck %s
+
+// CHECK-LABEL: func @check_success
+func @check_success() {
+    // CHECK: %[[CODE:.+]] =
+    %statusCode = constant 1 : i32
+    // CHECK: vm.cond_fail %[[CODE]], %[[CODE]]
```

Comment:
Thinking about adding some sugar to simplify this case to just `vm.cond_fail %[[CODE]]`, and the case below to `vm.cond_fail %[[CODE]], "failure message"`